internal/query/project_test.go

```diff
@@ -0,0 +1,371 @@
+package query
+
+import (
+	"database/sql"
+	"database/sql/driver"
+	"errors"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/caos/zitadel/internal/domain"
+	errs "github.com/caos/zitadel/internal/errors"
+)
+
+var (
+	now = time.Now()
+)
+
+func Test_prepareProjectsQuery(t *testing.T) {
+	type want struct {
+		sqlExpectations sqlExpectation
+		err             checkErr
+	}
+	tests := []struct {
+		name    string
+		prepare interface{}
+		want    want
+		object  interface{}
+	}{
+		{
+			name:    "prepareProjectsQuery no result",
+			prepare: prepareProjectsQuery,
+			want: want{
+				sqlExpectations: mockQueries(
+					`SELECT zitadel\.projections\.projects\.id,`+
+						` zitadel\.projections\.projects\.creation_date,`+
+						` zitadel\.projections\.projects\.change_date,`+
+						` zitadel\.projections\.projects\.resource_owner,`+
+						` zitadel\.projections\.projects\.state,`+
+						` zitadel\.projections\.projects\.sequence,`+
+						` zitadel\.projections\.projects\.name,`+
+						` zitadel\.projections\.projects\.project_role_assertion,`+
+						` zitadel\.projections\.projects\.project_role_check,`+
+						` zitadel\.projections\.projects\.has_project_check,`+
+						` zitadel\.projections\.projects\.private_labeling_setting,`+
+						` COUNT\(\*\) OVER \(\)`+
+						` FROM zitadel\.projections\.projects`,
```

Comment:
do we need regex queries or could we just pass a normal string and create a regex from that using `regexp.QuoteMeta` in the mockQueries function?

Reply:
btw. when using single ticks, why don't you just put in a single string without `+`:
```
`SELECT zitadel\.projections\.projects\.id,
	zitadel\.projections\.projects\.creation_date,
	zitadel\.projections\.projects\.change_date,
	...
```

Reply:
this would result in a different string. => with new lines